### DISCUSSION TITLE
Finally

### DISCUSSION THREAD

**Discussion by thenicnet** [OMOTE-Community/OMOTE-Firmware#2] [General]
I needed to let you know, I found this via a twitter post and I'm so excited.

I'm currently in the middle of a project, but this is next up on the list.  My whole house runs on Logitech Harmony remotes and I'm really just waiting for the day they stop working to have to figure out the next solution. 

Thank you for putting this together and I hope eventually I can contribute to it in some way.

**Comment by Syco54645** [comment]
> I needed to let you know, I found this via a twitter post and I'm so excited.
> 
> I'm currently in the middle of a project, but this is next up on the list.  My whole house runs on Logitech Harmony remotes and I'm really just waiting for the day they stop working to have to figure out the next solution. 
> 
> Thank you for putting this together and I hope eventually I can contribute to it in some way.

I feel the same way. I dread the day my harmony hub dies. Looking forward to this project.

**Comment by CoretechR** [comment]
Thank you! There are quite a lot of people looking for a replacement for their harmony or an open-source remote in general.
It would be awesome to see others build my remote. The software is in a state right now where it works for my personal devices but it's really missing a sort of configurator. Maybe a web interface that is hosted on the ESP32 itself.

**Reply by riker09** [reply]
I never felt the need to aquire a universal remote like a Harmony before. That was until now.

My home runs mainly in Home Assistant. And yes, there should be a way to integrate your remote with it. MQTT, REST API, ... I just discovered this project and decided to make IT the top priority. What I'd really hope for is a whole kind of eco system where it is really easy to add new things (screens) to your remote. Maybe something simple like the YAML configuration with ESPHome?

Also, I'd pay money for an assembled board and/or casing as many other people have stated before.

This remote is a real game changer, I really wish for it to be a huge success.

**Reply by SIR-Inc** [reply]
I was thrilled to find out about this project from the [BeardedTinker](https://www.youtube.com/watch?v=lom9cs3mI4A) YouTube channel. I have decades of experience with home automation and as an electronics engineer I develop a lot of my own platforms and integrate them with [Home Assistant](https://www.home-assistant.io). I've seen a few comments about support for Home Assistant and I strongly suggest leveraging a relationship with Home Assistant. There are a ton of people who can help and support the OMOTE project and if the OMOTE hardware is compatible or works well with Home Assistant there is a potential market for the sale of a lot of devices and or kits. People have already posted comments about buying a kit or assembled remotes and I agree that there could be a lof people looking for a solution like this. I am one of those people who has been unsuccessfully looking for a viable solution to my remote control hardware needs in my Home Assistant project. There are a few projects and devices currently available but they are all prohibitively expensive and I feel they all make the mistake of having too many features. I would also support a basic remote option that only has essential buttons, no screen and only uses Zigbee. Everyone seems to think that they have to develop a really sophisticated and fancy solution when in fact the real need is for a simple, cheap and reliable device. Take a look at most of the remote options for Google, Apple and Fire TV, they are mostly basic remotes with a few buttons and use radio frequency [I'm ignoring that some have track pads or voice recognition]. I've just printed my first OMOTE case and will be ordering PCB's shortly. I wish all the very best for the OMOTE project. Thank you!

**Reply by CoretechR** [reply]
Thanks for mentioning the video, @SIR-Inc ! I'll share it on the Discord.
I have thought about making an Omote Light at some point, too. Reduced BOM count and no display would make this much easier to source and assemble. What interfaces should it have in your opinion? IR and Zigbee only?

**Reply by SIR-Inc** [reply]
I feel the need to preface my reply by saying I fully appreciate that there are countless scenarios where different people have set up different equipment with different technologies/services/vendors and this makes it very challenging to come up with solutions that will make the majority of people happy. There will always be people who have a bizarre, one of a kind system or configuration and need a custom solution which is why I strongly advocate for modular, standards based platforms that will almost always accommodate every persons requirements. This is the reason I am so happy with Home Assistant and strongly support it. Home Assistant has come a long way and while it is not perfect and has a little more 'growing up' to do it checks the most boxes for the most people and I believe has the potential to evolve into a near perfect solution for all of us. Having said all that I really like the idea of an OMOTE Light and my personal preference would be to have the device only use Zigbee but I begrudgingly acknowledge that there are so many people who need to control devices via IR that I think the best option that would work for the most people would require a device with both Zigbee and IR. I would keep the software very simple by flashing the ESP32 SOC with [ESPHome](https://esphome.io/). Obviously I would be doing this with the intention of integrating the device into Home Assistant but even if the device is used standalone you only need a USB interface and a text editor to edit the configuration (YAML). Once the initial configuration is flashed you can then do the rest OTA. I think it's amazing that there is the option to enable a built-in web server so the device can have a web page for control and or monitoring. With knowledge of how the remote's buttons, IR emitter etc. are interfaced to the ESP32 (GPIO, I2C etc.) it would take less than an hour to come up with a YAML configuration file that could be shared. This is very basic and manual but I see this as the quickest, simplest, cheapest and most robust way to quickly get a working device operational. This also makes it an almost blank slate that people can customize to suit their projects. I find that simply getting something basic to work helps motivate me to then build out more complex features and functionality. You know the expression: Aim low to avoid disappointment. I've read a few posts where people have commented that it required a lot of effort to get all the parts and then assemble the device only to find it didn't work. An experience like this usually results in the project ending up on a back shelf destined to stay there and collect dust. Learn to walk before you run, okay I'll stop with the expressions. Just my thoughts regarding how an OMOTE Light might play out.

**Reply by SIR-Inc** [reply]
A quick post about a curiosity of mine. The thought crossed my mind about being able to source a generic, remote shell without electronics. Billions of TV remotes are manufactured so I was wondering if it is at all viable to find a shell with suitable buttons and then design a PCB, battery etc. to fit. I have a good 3D printer and the case I printed looks and feels great to me and I don't have a problem with the 3D printed case approach but not everyone has a 3D printer and I've seen a lot of critical comments about the quality, look, functionality and feel of 3D printed cases. Perhaps this could be a solution to some of these challenges. This is only an idea that I just thought of and I haven't done any research. Perhaps this has already been thought of but I wanted to share my thoughts just in case.

**Reply by CoretechR** [reply]
Actually, Pete on the Discord channel made a whole custom pcb and used the silicone keypad of an off-the-shelf remote as buttons. So this is definitely something that can be done. I just could not find a remote that was widely available so I went with the 3D-printed case. There are hundreds of regular IR remotes, but the designs seem to change often and you never know when they stop being manufactured.

**Reply by SIR-Inc** [reply]
All good points. Thanks for the feedback. I'm happy with my options to 3D print. For those of us who have lots of equipment, tools, expertise and knowledge it's not a problem but I try to be considerate of the many people who don't have access to the resources I do and try come up with viable options. Not easy.

I think I need to dust off my Discord credentials and go take a look.
